include/sys/range_tree.h

```diff
@@ -48,6 +48,32 @@ typedef enum zfs_range_seg_type {
 	ZFS_RANGE_SEG_NUM_TYPES,
 } zfs_range_seg_type_t;
 
+/*
+ * Range tree behavior flags.
+ *
+ * The UC (use case) flags are intended to support the zfs_recover_rt mode.
+ * The range tree's logic needs to know the context in order to correctly
+ * recover from an unexpected situation by exchanging potential data loss for
+ * a potential space leak:
+ *
+ * - If it knows that the tree represents allocated space then it should better
+ *   perform an unexpected addition to the tree.
+ *
+ * - Similarly, if it's about free space (aka allocatable) then it should
+ *   perform unexpected removals instead of silently ignoring the issue.
+ *
+ * The generic case means to simply ignore unexpected additions/removals as
+ * a recovery mechanism, without special treatment.
+ *
+ * Unexpected actions are logged with extra details such as a range tree
+ * name string, which can be marked as dynamic to be freed along with the tree
+ * instance destruction.
+ */
+#define	ZFS_RANGE_TREE_F_UC_GENERIC		(1 << 0)
```

Comment:
I don't think "GENERIC" is really meaningful. Easier and cleaner I think would be just to pass 0 if we can't say anything better (we really should).